src/basic/log.h

```diff
@@ -73,6 +73,9 @@ int log_get_max_level_realm(LogRealm realm) _pure_;
  * for the application itself.
  */
 
+assert_cc(STRLEN(__FILE__) > STRLEN(RELATIVE_SOURCE_PATH) + 1)
+#define _FILE_ (__FILE__ + STRLEN(RELATIVE_SOURCE_PATH) + 1)
```

Comment:
Hmm, this is supposed to cut the leading path off?

Can we maybe name this differently? i.e _PROJECT_FILE_ or _FILE_RELATIVE_TO_PROJECT_ or so?

This looks super fragile to me, btw, can't we somehow detect discrepancies between the prefix and the __FILE__ thin during compilation? maybe at least:

```c
assert_cc(__FILE__[0] == RELATIVE_SOURCE_PATH[0]);
assert_cc(__FILE__[1] == RELATIVE_SOURCE_PATH[0]);
assert_cc(__FILE__[STRLEN(RELATIVE_SOURCE_PATH)] == '/');
assert_cc(__FILE__[STRLEN(RELATIVE_SOURCE_PATH)-1] == RELATIVE_SOURCE_PATH[STRLEN(RELATIVE_SOURCE_PATH)-1]);
```

you get the idea?

And I must say I really wished meson had a concept for that, i.e. so that it could generate the tree-relative source path automatically and pass it to each gcc invocation right-away with -D, so that we don't have to do such tricks here

Reply:
Right, PROJECT_FILE sounds good. I'll reword this part.

Reply:
I looked into adding more tests using `assert_cc`, but I forgot that there's already a test in test-log, simply using startswith. So that should be enough.